libsignal-service/src/sender.rs

```diff
@@ -645,23 +646,20 @@ where
                 )
                 .await?,
             );
+        } else {
+            devices.push(DEFAULT_DEVICE_ID);
         }
 
-        let mut devices = vec![DEFAULT_DEVICE_ID];
         devices
             .extend(recipient.sub_device_sessions(&self.session_store).await?);
+        devices.dedup();
 
         // When sending to ourselves, don't include the local device
         if myself {
             devices.retain(|id| *id != self.device_id);
         }
 
         devices.sort_unstable();
-        let original_device_count = devices.len();
-        devices.dedup();
-        if devices.len() != original_device_count {
-            log::warn!("SessionStoreExt::get_sub_device_sessions should return unique device id's, and should not include DEFAULT_DEVICE_ID.");
-        }
```

Comment:
I thought this would be a useful warning/runtime check. What's the reason behind removing it?

Reply:
I think it got lost somewhere along the way... It's a useful check to have, I'll put it back.